search-improvements.rst

```diff
@@ -0,0 +1,189 @@
+.. post:: August 14, 2018
+   :tags: gsoc, search, feature
+   :author: Safwan
+   :location: DHA
+
+Improved Search
+==================================
+Have you ever struggled with a poorly documented software project?
+What about a well documented project but you can't find the right section inside the docs?
+The Read the Docs `core team`_ has realized the importance of good search for documentation
+and got me to take the challenge as a Google Summer of Code student.
+The main goal of my GSoC project was to refactor the search code together with upgrading the backend
+search engine, as well as adding more features to our search functionality like exact match search,
+case insensitive search, search as you type, suggestions and more.
+
+Google Summer of Code
+^^^^^^^^^^^^^^^^^^^^^
+Google Summer of Code is a global program where students work with an open source organization
+on a 3 month programming project. The core team of Read The Docs proposed some `Project Ideas`_,
+one of them was **Refactor & improve our search code**. I (`Safwan Rahman`_) was keen to get my hand dirty with
+Elasticsearch_ and grasped the opportunity to do so by applying
+for this project and I got accepted.
+
+I have worked full time for last 3 months to upgrade the whole codebase
+to compatible with `Elasticsearch 6.x`_ and also implemented various features like:
+
+- `Exact Matching Search`_
+- `Case Insensitive Search`_
+- `Improved Result Order`_,
+- `Zero Downtime Indexing`_
+
+Together with this, I am willing to implement more features like following:
+
+- `Search as You Type and Autocomplete`_
+- `Code Search`_ 
+
+All of my search related work can be seen in the `Search Project Board`_.
+
+
+Background
+^^^^^^^^^^
+Search is a vital part of any documentation hosting platform, so people can get the
+information they need. As a documentation hosting platform, the same rules apply to
+Read The Docs. Because of having a small core team, the search functionality
+of Read The Docs has lagged behind for quite a while now. Initially the search code
+was voluntarily contributed by `Rob Hudson`_,  `back in 2013`_ and then improved by other
+contributors. The search infrastructure was already outdated as
+Read The Docs were using `Elasticsearch 1.3.x`_ which was already reached its
+End of Life in 2016. Therefore, the upgradation of search infrastructure was badly needed.
```

Comment:
Therefore, the upgradation of search infrastructure -> Upgrading the search infrastructure